express/src/middleware/mobileValidator.spec.ts

```diff
@@ -0,0 +1,142 @@
+   });
+ 
+   it('a valid +4479 number is accepted. ', async function () {
+       assert.equal(onlyNumbers(VALID_NUMBER_STARTING_WITH_PLUS_44), true, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44} to be allowed because it only contains numbers and the plus symbol`);
+   });
+ 
+   it('a number which contains a letter is not accepted. ', async function () {
+       assert.equal(onlyNumbers(NUMBER_WITH_BAD_CHARACTERS), false, `Expected ${NUMBER_WITH_BAD_CHARACTERS} to not be allowed because it only contains the letter O`);
+   });
+});
+ 
+ 
+describe('Checking that spaces and dashes within a number are removed ', () => {
+ 
+   it('A valid 07 number with spaces will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_0_CONTAINING_SPACES), VALID_NUMBER_STARTING_WITH_0, `Expected ${VALID_NUMBER_STARTING_WITH_0_CONTAINING_SPACES} to be ${VALID_NUMBER_STARTING_WITH_0}`);
+   })
+ 
+   it('A valid +44 number with spaces will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_SPACES), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_SPACES} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+ 
+   it('A valid 07 number with dashes will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_0_CONTAINING_DASHES), VALID_NUMBER_STARTING_WITH_0, `Expected ${VALID_NUMBER_STARTING_WITH_0_CONTAINING_DASHES} to be ${VALID_NUMBER_STARTING_WITH_0}`);
+   })
+ 
+   it('A valid +44 number with dashes will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_DASHES), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_DASHES} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+   it('A valid 07 number without spaces will not be cleaned ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_0), VALID_NUMBER_STARTING_WITH_0, `Expected ${VALID_NUMBER_STARTING_WITH_0} to be ${VALID_NUMBER_STARTING_WITH_0}`);
+   })
+ 
+   it('A valid +44 number without spaces will not be cleaned . ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+})
+ 
+describe('Checking that only numbers starting with +447[1-9] and 07[1-9] are accepted', () => {
+   it('An invalid +48 number is rejected.', async function () {
+       assert.equal(await ukNumberCheck(INVALID_NUMBER_STARTING_WITH_PLUS_48), false, `${INVALID_NUMBER_STARTING_WITH_PLUS_48} should not be allowed as it does not begin with +447`);
+   })
+ 
+   it('A valid +44 number is accepted.', async function () {
```

Comment:
This needs clarifying - it says it tests that a valid +44 number is accepted, tests a number starting +4470 (which should not be accepted as valid - use a different example number for this method if necessary) and if the test were to fail it will report that "+4470whatever should be allowed as it does not begin with +447".  Except it does begin with +447.